src/plugins/validate-semantic/validators/form-data.js

```diff
@@ -49,41 +49,43 @@ export const validateParameterFormDataForFileTypes = () => system => {
 
 export const validateParameterFormDataConsumesType = () => system => {
   return system.validateSelectors
-    .allParameters()
+    .allPathItems()
     .then(nodes => {
       return nodes.reduce((acc, node) => {
         const value = node.node
+        const globalConsumes = getRootNode(node).node.consumes ||[] 
+        var parameters = value.parameters
 
-        const globalConsumes = getRootNode(node).node.consumes ||[]
+        var isThereAnyFormData = parameters != null && parameters.find(parameter => parameter.in === "formData")
+        var isThereAnyFile = parameters != null && parameters.find(parameter => parameter.type === "file")
 
-        // step out of parameter item and array
-        const containingContext = node.parent.parent.node
-        const contextConsumes = containingContext.consumes || []
-
-        const combinedConsumes = [...globalConsumes, ...contextConsumes]
-        if(
-          value.type === "file" &&
-          value.in === "formData" &&
-          combinedConsumes.indexOf("multipart/form-data") === -1
-        ) {
-          acc.push({
-            message: `Operations with parameters of "type: file" must include "multipart/form-data" in their "consumes" property`,
-            path: [...node.path],
-            level: "error",
-            source: SOURCE
-          })
-        } else if(
-          value.in === "formData" &&
-          combinedConsumes.indexOf("multipart/form-data") === -1 &&
-          combinedConsumes.indexOf("application/x-www-form-urlencoded") === -1
-        ) {
-          acc.push({
-            message: `Operations with Parameters of "in: formData" must include "application/x-www-form-urlencoded" or "multipart/form-data" in their "consumes" property`,
-            path: [...node.path.slice(0, -2)],
-            level: "error",
-            source: SOURCE
-          })
+        for (var method of operationKeys) {
+          if (value[method] != null) {
+            parameters = value[method].parameters 
```

Comment:
**Change request**: I'd prefer to create two `const`s (`operationParameters` here, and `pathItemParameters` on line 57) rather than reusing a `parameters` variable, unless there's a specific reason to follow this pattern here (please share if this is the case!).

Constants [are easier to reason about](https://medium.com/dailyjs/use-const-and-make-your-javascript-code-better-aac4f3786ca1) 🙂

Reply:
Nice!